include/oneapi/dpl/experimental/kt/esimd_radix_sort_onesweep.h

```diff
@@ -581,11 +581,17 @@ onesweep(sycl::queue __q, _Range&& __rng, ::std::size_t __n)
     constexpr uint32_t GLOBAL_OFFSET_SIZE = BINCOUNT * STAGES * sizeof(global_hist_t);
     size_t temp_buffer_size = GLOBAL_OFFSET_SIZE + SYNC_BUFFER_SIZE;
 
-    uint8_t *tmp_buffer = sycl::malloc_device<uint8_t>(temp_buffer_size, __q);
+    const size_t full_buffer_size_part1 = temp_buffer_size * sizeof(uint8_t);
+    const size_t full_buffer_size_part2 = __n * sizeof(KeyT);
+    const size_t full_buffer_size = full_buffer_size_part1 + full_buffer_size_part2;
```

Comment:
The variables can be removed without any loss of expressiveness. Could you extend `temp_buffer_size`: 
```
size_t temp_buffer_size = GLOBAL_OFFSET_SIZE + SYNC_BUFFER_SIZE + __n * sizeof(KeyT);
```

Reply:
As for me this variable still useful and we shouldn't change it.

Reply:
I see that part1 and part2 variables are used separately, so this comment does not make any sense.